draft-irtf-cfrg-bls-signature.md

```diff
@@ -1083,6 +1102,114 @@ Procedure:
 6. return CoreVerify(PK, message, signature)
 ~~~
 
+# Interactive BLS Signatures {#interactive-bls}
+
+This section describes an private, interactive protocol between signer (server)
+and verifier (client) for computing the BLS signature variants in {{schemes}}. 
+At a high level, the signer and verifier interact to compute a blind signature
+protocol yielding `signature = Sign(SK, message)`, where `msg` is the private
+message to be signed,  and `SK` is the server's private key. In this protocol,
+the signer learns nothing of `message`, whereas the verifier learns `signature`
+and nothing of `SK`.
+
+This protocol roughly runs as follows:
+
+~~~
+   Verifier(PK, message)                       Signer(SK)
+  -------------------------------------------------------
+   blind_msg, R = Blind(PK, message)
+
+                         blind_msg
+                        ---------->
+
+                     blind_sig = BlindSign(SK, blind_msg)
+
+                         blind_sig
+                        <----------
+
+  signature = Finalize(PK, message, R, blind_sig)
+~~~
+
+Upon completion, correctness requires that clients can verify signature `signature`
+over private input message `message` using the server public key `PK` by invoking
+the Verify routine. The finalization function performs that check before returning
+the signature.
+
+## Blind {#proto-blind}
+
+The Blind function takes as input the signer public key `PK` and message to sign
+`message` and produces a blinded message and blind inverse.
+
+~~~
+signature = Blind(PK, message)
+
+Inputs:
+- PK, a public key in the format output by SkToPk.
+- message, an octet string.
+
+Outputs:
+- R, a scalar in G1.
```

Comment:
What does "in G1" mean? That blind signatures make sense only over E1? If so, then it's arguably not optimal spot for such statement. And if not, then it's a number less than `r` irregardless the curve. BTW, should one specify it as `1 < R < r`? More specifically "more than 1" to ensure that `1/R` != `R`?

Reply:
Hmm, I was assuming that this would only work over E1, though maybe that constraint is not strictly necessary?

Reply:
(Also, separately agreed on the constraint for R :))

Reply:
> this would only work over E1

Algebraically it's fully symmetric. I mean it does work either way.

Reply:
Sorry, I mean that the _change_ constrained itself to one curve, not that the mechanism itself doesn't work across both.